src/base/coordinate-grid-mixin.js

```diff
@@ -503,11 +546,12 @@ export class CoordinateGridMixin extends ColorMixin(MarginMixin) {
         }
 
         let axisXLab = g.select(`text.${X_AXIS_LABEL_CLASS}`);
+        const axisXLabY = (this._useTopXAxis ? 0 : this.height()) +
+            this._xAxisLabelPadding * (this._useTopXAxis ? 1 : -1);
```

Comment:
I think it would be clearer to write this as

 ```suggestion
         const axisXLabY = this._useTopXAxis ? this._xAxisLabelPadding : (this.height() - this._xAxisLabelPadding);
```